### PR TITLE
Update to Electron 16

### PR DESCRIPTION
**Pull request by kicken**
Re-implement tray envelope icon for unread messages.

Fix minimize on close bug.

### CODE REVIEW

**Comment by jcbrand** [comment]
@kicken: I haven't forgotten this PR, will look at it once I'm ready to release Converse 9.0.1 which is hopefully soon.

**Comment by jcbrand** [comment]
@kicken: I wanted to merge this MR now, but when I merge it and run `make serve` to get the electron build, I get the following error in the developer console:

`Uncaught (in promise) Error: Error invoking remote method 'settings': No handler registered for 'settings'`

And Converse doesn't appear, I just see a blank page.

File: app/credentials.js [hunk 1]
```diff
@@ -1,48 +1,32 @@
...
-const settings = require('electron-settings');
-const keytar = require('keytar')
-
-function addCredentials (connectionManager, login, password) {
+async function addCredentials(connectionManager, login, password) {
     const xmppService = login.split('@').pop()
-    settings.setSync('connectionManager', connectionManager)
-    settings.setSync('login', login)
-    keytar.setPassword(xmppService, login, password)
+    await api.settings.set('connectionManager', connectionManager)
```
Comment (jcbrand) [inline app/credentials.js:5]: These calls cause the following error for me:

`Uncaught (in promise) Error: Error invoking remote method 'settings': No handler registered for 'settings'`

Comment (jcbrand) [reply app/credentials.js:5]: Looks like moving the `icpMain.handle` lines higher up in main.js solves the issue...

**Comment by kicken** [comment]
@jcbrand it seems like the problem was using a path to a missing icon in the tray service.  I've corrected the code to use the 16x16 image and it seems to work properly for me now on my mac.

**Pull request merged**